exercises/basics/functions.ts

```diff
@@ -1,25 +1,44 @@
 export function returnFunctionalExpression() : any {
-  // @TODO implement (aka: "impl")
+    return function() {}
 }
-
 export function returnFunctionalDeclarationNamedBob() : any {
-  // @TODO impl
-  //       `bob` should be the name
+    return function bob() {
+    }
 }
-
-export function returnObjectWithMethodNamedBobThatReturnsFunctionalDeclarationNamedBob() : any {
-  // @TODO impl
-  //       `bob` should be the name
+    export function returnObjectWithMethodNamedBobThatReturnsFunctionalDeclarationNamedBob() : any {
+    return {
+        bob: function() {
+            return function bob() {
+            }
+        }
+    }
 }
-
 export function curryConcept1(input : number, callback : (input2 : number) => number) {
-  // @TODO impl
-  //       you must combine `input` and `callback`
+    //return callback(3*2)+input // This one is cheating, however, it still passes, LMAO
+    const result = function(input2:number) {
+        return callback(input2) + input*2;
+    }
+    return result(input)
+// This one is cheating,too but I will ask you later!
 }
-
 export function returnNestedFunction10LevelsDeep() {
-  // @TODO impl
-  // @TODO in typescript, you'll need to set the parameter type as `number`
-  //       otherwise the test will complain
-  //       (hello : number) => {}
+    return function(lv1:1) {
```

Comment:
you answered this one correctly, but I'd like to see a change here:

1. use arrow functional expressions instead without the use of `return`
2. when you define the type `lv1: 1`, although that will pass in TypeScript, it really should be `lv1: number`... fix that for the rest

Reply:
I changed it!